### DISCUSSION TITLE
Review

### DISCUSSION THREAD

**Discussion by NazgulM** [lab-brussels-1/All-About-Trees-Group-3#16] [Announcements]
Dear Group,
Please ask here or give us some advices.

**Comment by alexanderism55** [comment]
I would love to know what git rebase does exactly

**Reply by sammou00** [reply]
@alexanderism55  ok , `git rebase`  
let us say we have a master branch and we make some changes then add them and commit them , then we create a new branch called `branch-1` , then we work on both master and `branch-1`  now we have new commits on the master and also `branch-1` 
 now if I am on the master branch  : 
 and type `git rebase branch-1` git will go to branch-1 and master it will see the last commit both master and `branch-1` have in common which will be the one before creating `branch-1` then it will compare the changes on  both and kind of get the new changes from `branch-1` . to master , you will also have the commit from `branch-1` to master 
 if you are on `branch-1` and do git rebase master  : 
same idea git will compare both branches and kind of git the changes from master to branch-1 and you will have the master commits on  branch `branch-1` 

so the base commit which was on both master and branch-1 before rebase was the one before creating the branch 
now after rebase I will need to add a new commit which will be the new last common commit between master and branch-1

**Reply by sammou00** [reply]
Here is a video https://www.youtube.com/watch?v=CRlGDDprdOQ&t=855s

**Comment by alexanderism55** [comment]
## I get it now
**It basically helps you get updates from branches locally**. It has similarity with the act of pulling from the origin before pushing to get the latest version of the master branch. Except that with git rebase; `branch one` will adapt the new changes in `branch two` and `branch two` will do the same. So both branches will be up-to date with each other.
***THANKS SAMIR***

**Reply by sammou00** [reply]
same example 
master , if i do git rebase branch-1 , the mater will have all the new commit and change from branch-1 , but branch-1 won't 
branch-1 same here git rebase master , this branch-1 will have the master commits and change , but the master no 
I think the video is explain it well .

**Comment by sammou00** [comment]
![image](https://user-images.githubusercontent.com/67711948/159567570-3f0bbc14-b9e0-453a-8d3f-44e7b6982ed5.png)

**Reply by sammou00** [reply]
I really like the way your design your user-personas it is really  nice , well done. but do you think that everyone is inters ted about tress or reading about tress ?  maybe you can define more specific users like  environmental-activist. you can keep it like that , but normally user-personas should be more specific than `Public` everyone.
if you defined your user personas more specifically it will help you to build a better website according to his/her needs.
I will talk about that next Sunday too. But well done.

**Comment by sammou00** [comment]
Example from group 4 , it could be interesting
![image](https://user-images.githubusercontent.com/67711948/159569053-d8b883ed-12f0-4208-b179-eaf7bcf3b322.png)

**Comment by sammou00** [comment]
Hello, very good job we all are proud of you 

Review 👍 

1- For user-personas : `the public` is a very general target , you should be more specific (see group1 discussion)
[group 1](https://github.com/lab-brussels-1/The-first-group/discussions/29) about this point.

2- Very nice `backlog`

3- Nice development- strategy 

4- Nice design again for pictures or images we use also motioned in group 1 discussion
![image](https://user-images.githubusercontent.com/67711948/159983749-74b0ca84-db6b-4006-b21b-ed2e90c4c054.png)

**Comment by sammou00** [comment]
5 -No comments in `html` and `CSS` why ? comments are very important 

![image](https://user-images.githubusercontent.com/67711948/159984215-e021bc43-c532-4231-b887-a48d58e12a93.png)
![image](https://user-images.githubusercontent.com/67711948/159984288-0d927502-aea3-4223-982a-d0a5128a8c96.png)

**Reply by NazgulM** [reply]
Thank you for your detailed review, that is very helpful for us. Do we need change all of these issues?

**Reply by sammou00** [reply]
@NazgulM  No, those for the next week , just keep them in mind for the next project

**Reply by MadhuMNG** [reply]
Hi Samir,

Thanks for your review.

I need your help on the below error.

![Screenshot 2022-03-24 at 20 27 42](https://user-images.githubusercontent.com/100073498/159995122-f44d1c46-724d-43e6-a61d-5c606cc3749c.png)

**Reply by sammou00** [reply]
Ok , you can't use the same id , id must be unique
